x-pack/elastic-agent/pkg/agent/program/testdata/single_config.yml

```diff
@@ -74,6 +74,23 @@ inputs:
   host: "http://localhost:80/service/status"
   timeout: 16s
   wait: 1s
+- type: network/flow
+  name: flow-1
+  timeout: 10s
+  period: 10s
+  keep_null: false
+  streams:
+    - type: network/flow
+      data_stream:
+        dataset: network.flow
+        type: logs
+- type: network/icmp
```

Comment:
Do we even need the `streams` part? Is it possible to have multiple streams inside `network/icmp`?

Reply:
AFAIK `streams` are required for all packages to function and are used in determining some parts of event payloads that only agent configuration knows about, things like [`data_stream.namespace`](https://github.com/elastic/beats/blob/cffc81db9ae1f6ceab7c61ba5407e517165d1414/x-pack/elastic-agent/pkg/agent/transpiler/rules.go#L651) and [index specification](https://github.com/elastic/beats/blob/cffc81db9ae1f6ceab7c61ba5407e517165d1414/x-pack/elastic-agent/pkg/agent/transpiler/rules.go#L574). Ideally we wouldn't need these in the case where you only have one possible stream input, but I believe that's how the spec works. If I'm misunderstanding that though, let me know.

Reply:
You can also define the streams part directly on the input level. Or in the case of packetbeat, it could be defined potentially directly in the go code. This applies in all cases where the type and dataset are fixed and should not be user configurable.